noise2same/network.py

```diff
@@ -709,32 +752,39 @@ def forward(self, x: T) -> T:
 
         x = self.bottom_block(x)
 
-        for (
-            i,
-            (upsampling_block_local, upsampling_block_global, decoder_block, skip),
-        ) in enumerate(
-            zip(
-                self.upsampling_blocks_main,
-                self.upsampling_blocks_secondary,
-                self.decoder_blocks,
-                encoder_outputs[::-1],
-            )
+        for (i, (upsampling_block, decoder_block, skip),) in enumerate(
+            zip(self.upsampling_blocks, self.decoder_blocks, encoder_outputs[::-1],)
         ):
-
-            if self.unit_type == "dual_pass":
-                x_l, x_g = x
-                x_l = upsampling_block_local(x_l)
-                x_g = upsampling_block_global(x_g)
-
+            x = upsampling_block(x)
+            if self.unit_type == "dual_pass":  # when the skip is tuple
+                (x_l, x_g) = x if type(x) is tuple else (x, 0)
                 if self.skip_method == "add":
+                    # enc dec paramterization not supported
                     x_l.add_(skip[0])
                     x_g.add_(skip[1])
                 elif self.skip_method in ("cat", "concat"):
-                    x_l = torch.cat([x_l, skip[0]], dim=1)
-                    x_g = torch.cat([x_g, skip[1]], dim=1)
-                    x = (x_l, x_g)
+
+                    if self.ffc_enc == True and self.ffc_dec == False:
+                        combined_skip = torch.cat([skip[0], skip[1]], dim=1)
+                        x = torch.cat([x, combined_skip], dim=1)
+                    elif (
+                        self.ffc_enc == False and self.ffc_dec == True
+                    ):  # probably not worth it
+                        n, c, *s = skip.shape  # s is (d,h,w) or (h,w)
+                        global_channels = int(c * self.global_ratio)
+                        local_channels = c - global_channels
+                        skip_split = torch.split(
+                            skip, [local_channels, global_channels], dim=1
+                        )  # channel dimension
+                        x_l = torch.cat([x_l, skip_split[0]], dim=1)
+                        x_g = torch.cat([x_g, skip_split[1]], dim=1)
+                        x = (x_l, x_g)
```

Comment:
I agree that we can skip this evaluation for now, but would be nice to have a function `split_channels()` here in future